### PR TITLE
search: allow filtering by existence of store path

### PR DESCRIPTION
**Pull request by cdmistman**
Why
===

allows filtering results by whether the store path exists

What changed
============

- added `--filter-built` flag
- optionally filter fuzzy search results by presence of the store path

Test plan
=========

1. `nix eval github:replit/rippkgs/cad/filter-available#lib.genRegistry --refresh --apply 'f: f builtins.currentSystem (import <nixpkgs> { config = { allowUnsupportedSystem = true; }; })' --impure --json >nixpkgs.json`
2. `nix run github:replit/rippkgs/cad/filter-available#rippkgs-index -- --output nixpkgs.sqlite --registry nixpkgs.json`
3. `nix run github:replit/rippkgs/cad/filter-available#rippkgs -- --filter-built --index nixpkgs.sqlite <something>`
  - if you put `<something>` that you've previously built from the specific nix channel, then the result should appear
  - if you put `<something>` you haven't built, then the result should not appear. but if you remove the `--filter-built` flag, then the result should appear!

### CODE REVIEW

File: src/bin/search/fuzzy.rs [hunk 1]
```diff
@@ -23,20 +29,39 @@ pub fn search(query_str: &str, db: &Connection, num_results: u32) -> eyre::Resul
 SELECT *
 FROM packages
 ORDER BY fuzzy_score(name, ?1) DESC
-LIMIT ?2
```
Comment (ryantm) [inline src/bin/search/fuzzy.rs:26]: Isn't this going to slow things down a lot as it returns every package in the index?

Comment (cdmistman) [reply src/bin/search/fuzzy.rs:26]: nope! afaict rusqlite lazily fetches the rows (and either way, only the rows that are needed are processed in the functions below, due to lazy iterators). with `time -p` i'm not noticing any material difference in runtime :)

Comment (cdmistman) [reply src/bin/search/fuzzy.rs:26]: > any material difference

before: ./target/debug/rippkgs -i ./nixpkgs.sqlite rustc  0.09s user 0.01s system 36% cpu 0.269 total

after: `./target/debug/rippkgs -i ./nixpkgs.sqlite rustc  0.11s user 0.02s system 82% cpu 0.158 total`

Comment (cdmistman) [reply src/bin/search/fuzzy.rs:26]: i do intend on soon adding `WHERE fuzzy_score > x` which should bring things back closer to the original timing (and possibly even better) but that's gonna happen in milestone 4

Comment (ryantm) [reply src/bin/search/fuzzy.rs:26]: Sweet, that's fancy.

**Review by ryantm — APPROVED** [review]

**Pull request merged**